lib/Data/Chronicle/Writer.pm

```diff
@@ -128,38 +140,80 @@ publish "category1::name1" in Redis if C<publish_on_set> is true.
 =cut
 
 sub set {
-    my $self     = shift;
-    my $category = shift;
-    my $name     = shift;
-    my $value    = shift;
-    my $rec_date = shift;
-    my $archive  = shift // 1;
-    my $ttl      = shift // $self->ttl;
+    my $self         = shift;
+    my $category     = shift;
+    my $name         = shift;
+    my $value        = shift;
+    my $rec_date     = shift;
+    my $archive      = shift // 1;
+    my $suppress_pub = shift // 0;
+    my $ttl          = shift // $self->ttl;
 
-    die "Recorded date is undefined" unless $rec_date;
-    die "Recorded date is not a Date::Utility object" if ref $rec_date ne 'Date::Utility';
-    die "Cannot store undefined values in Chronicle!" unless defined $value;
-    die "You can only store hash-ref or array-ref in Chronicle!" unless (ref $value eq 'ARRAY' or ref $value eq 'HASH');
+    $self->mset([[$category, $name, $value]], $rec_date, $archive, $suppress_pub, $ttl);
+
+    return 1;
+}
+
+=head2 mset
+
+Example:
+
+    $chronicle_writer->mset([["category1", "name1", $value1], ["category2, "name2", $value2], ...]);
 
-    $value = JSON::MaybeXS->new->encode($value);
+Store a piece of data "value1" under key "category1::name1", "category2::name2", etc in Pg and Redis. Will
+publish "category1::name1", "category2::name2", etc in Redis if C<publish_on_set> is true.
+
+=cut
+
+sub mset {
+    my $self         = shift;
+    my $entries      = shift;
+    my $rec_date     = shift;
+    my $archive      = shift // 1;
+    my $suppress_pub = shift // 0;
+    my $ttl          = shift // $self->ttl;
+
+    $self->_validate_value($_->[2]) foreach @$entries;
```

Comment:
If there's a lot of values to be set, and one of them fails validation, the error message won't mention the name of the one that failed. This probably makes debugging more difficult. Could you arrange to pass the name too? Maybe

    $self->_validate_value(@$_) foreach @$entries;

then it has access to the category and name too.